src/cmake-tools.ts

```diff
@@ -271,34 +287,52 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
       rollbar.invokeAsync('Changing CMake kit', async () => {
         if (kit) {
           log.debug('Injecting new Kit into CMake driver');
-          const drv = await this._cmakeDriver;
-          await drv.setKit(kit);
+          const drv = await this.getCMakeDriverInstance();
+          if(drv) {
+            await drv.setKit(kit);
+          }
         }
         this._statusBar.setActiveKitName(kit ? kit.name : '');
       });
     });
     this._ctestController.onTestingEnabledChanged(enabled => { this._statusBar.ctestEnabled = enabled; });
     this._ctestController.onResultsChanged(res => { this._statusBar.testResults = res; });
 
-    // Listen for terminal closing, we should create a new one on next launch
-    vscode.window.onDidCloseTerminal((term) => {
-      if (term == this._launchTerminal) {
-        this._launchTerminal = null;
-      }
-    });
-
-    // Finally, start the CMake driver
-    const drv = await (this._cmakeDriver = this._startNewCMakeDriver());
-    // Reload any test results. This will also update visibility on the status
-    // bar
-    await this._ctestController.reloadTests(drv);
     this._statusBar.setStatusMessage('Ready');
-    this._statusBar.targetName = this.defaultBuildTarget || await this.allTargetName;
 
     // Additional, non-extension: Start up nagging.
     this._nagManager.start();
   }
 
+  /**
+   * Returns, if possible a cmake driver instance. To creation the driver instance,
+   * there are preconditions that should be fulfilled, such as an active kit is selected.
+   * These preconditions are checked before it driver instance creation. When creating a
+   * driver instance, this function waits until the driver is ready before returning.
+   * This ensures that user commands can always be executed, because error criterials like
+   * exceptions would assign a null driver and it is possible to create a new driver instance later again.
+   */
+  async getCMakeDriverInstance() : Promise<CMakeDriver | null> {
+    if (!this._kitManager.hasActiveKit) {
+      return null;
```

Comment:
I think this should be returned as a `Promise`:
```ts
return Promise.resolve(null);
```

Reply:
Is there a coding style guide? Because I had it in different review and there I should replace it,

Reply:
I like both syntax promise.reject()/ resolve() and returns with throwing exceptions. The promise syntax makes it easier to see the interaction with a Promise. In the other case it is easier to change a async function to a sync function. 

Do you know a typescript coding style guide? If yes is it possible to run an automatic check for the style guide?

Reply:
I like both syntax promise.reject()/ resolve() and returns with throwing exceptions. The promise syntax makes it easier to see the interaction with a Promise. In the other case it is easier to change a async function to a sync function. 

Do you know a typescript coding style guide? If yes is it possible to run an automatic check for the style guide?